app/services/charts_layout_service.rb

```diff
@@ -34,6 +34,12 @@ def find_chart_path
       # First check if there is file for specific class of the @perf_record
       path = chart_path(@layout, @perf_record.class.name.gsub(/::/, '_'))
       return path if File.exist?(path)
+
+      # Fallback to check if there is a shared file for the family of classes
+      # similar to @perf_record
+      return chart_path(@layout, @perf_record.chart_layout_path) if
+        @perf_record.respond_to?(:chart_layout_path)
```

Comment:
@yaacov the idea was to avoid the costly `File.exist?(path)` check. And, after all... if it's specified we shouldn't try to guess it using the class name.